practice/tasks/IDisposable/IDisposableImplementation/IDisposableImplementation/Program.cs

```diff
@@ -1,22 +1,39 @@
 ﻿using System;
+using System.Diagnostics;
 
 namespace NetMentoring
 {
     internal class Program
     {
+        public static Stopwatch Timer { get; } = new Stopwatch();
+
         private static void Main(string[] args)
-        {            
-            for(var i = 0; i < 10000; i++)
-                WriteLog("Interation number #" + i);
+        {
+            Timer.Start();
+
+            for (var i = 0; i < 10000; i++)
+            {
+                if (i == 9999)
+                    WriteLog("Iteration number #" + i + " " + DateTime.Now.ToString("F"));
+                else
+                    WriteLog("Iteration number #" + i);
+            }
+            
+            var executionTime = Timer.ElapsedMilliseconds;
+            Timer.Stop();
 
             Console.WriteLine("Finished");
+            Console.WriteLine($"ExecutionTime: {executionTime} Milliseconds");
+            new MemoryStreamLogger().ReadFile();
```

Comment:
please compare this usage of MemoryStreamLogger and in the "WriteLog" message
Why do you call Dispose method only one time?
How do you know when it (Dispose) method should be called?

Reply:
I use using and it calls Dispose when we are going to leave this using